File: codewars/6kyu/iq-test/dduck-iq-test.py
# IQ Test
# https://www.codewars.com/kata/552c028c030765286c00007d

from unittest import TestCase


def iq_test(numbers):
    splited_numbers = list(map(int, numbers.split(" ")))
    is_even_number_more = len(list(filter(lambda x: x % 2 == 0, splited_numbers))) > 1

    if is_even_number_more:
        return [i for i, v in enumerate(splited_numbers) if v % 2 != 0][0] + 1

    return [i for i, v in enumerate(splited_numbers) if v % 2 == 0][0] + 1


TestCase().assertEqual(iq_test("2 4 7 8 10"), 3)
TestCase().assertEqual(iq_test("1 2 2"), 1)

File: codewars/6kyu/iq-test/dduck-iq-test.js
// IQ Test
// https://www.codewars.com/kata/552c028c030765286c00007d

const iqTest = (numbers) => {
  const splitedNumbers = numbers.split(" ").map(Number);
  const isEvenNumberMore = splitedNumbers.filter((v) => v % 2 === 0).length > 1;
  
  if (isEvenNumberMore) {
    return splitedNumbers.findIndex((v) => v % 2 === 1) + 1;
  }
  return splitedNumbers.findIndex((v) => v % 2 === 0) + 1;
};

console.log(iqTest("2 4 7 8 10"), 3);
console.log(iqTest("1 2 2"), 1);
console.log(iqTest("5 3 2"), 3);
